cmake/CheckPatch.cmake

```diff
@@ -0,0 +1,24 @@
+find_program(CHECKPATCH checkpatch.pl
+             HINTS ${PROJECT_SOURCE_DIR}/checkpatch)
+
+add_custom_target(checkpatch)
+set(MASTER_BRANCH "upstream/master")
+
+if(CHECKPATCH)
+    add_custom_command(TARGET checkpatch
+        COMMENT "Running checkpatch"
+        COMMAND
+            ${CHECKPATCH}
+            # Description of supported checks in
+            # https://github.com/tarantool/checkpatch/blob/master/doc/checkpatch.rst
+            --codespell
+            --color=always
+            --git ${MASTER_BRANCH}..HEAD
```

Comment:
Can it be a problem you compare against master branch? What will it report in case you didn't rebase?

Reply:
by default git ref is `origin/master` and this could be overridden with env variable `CHECKPATCH_GIT_REF`.